pyo3-derive-backend/src/defs.rs

```diff
@@ -30,6 +31,40 @@ impl Proto {
     {
         self.py_methods.iter().find(|m| query == m.name)
     }
+    // Since the order matters, we expose only the iterator instead of the slice.
+    pub(crate) fn setters(&self, implemented: HashSet<String>) -> SlotSetterIter {
+        SlotSetterIter {
+            iter: self.slot_setters.iter(),
+            implemented_protocols: implemented,
```

Comment:
I quite like `std::iter::from_fn`, and return `impl Iterator<Item = &'static str>` instead of making a new struct, but I am happy with this way too 👍

Reply:
Thanks, I didn't know that function.